### PR TITLE
Mandate password prompt for user creation

### PR DESCRIPTION
**Pull request by jamesreprise**
## What is the goal of this PR?

We now force passwords to be entered in a password prompt. This prevents passwords from showing up in the command history.

## What are the changes implemented in this PR?

We've changed the flow of the user create command. Now, when creating a user, you supply a username and are then prompted to supply a password separately.

### CODE REVIEW

File: common/Utils.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public static String readNonEmptyLine(LineReader reader, String prompt) throws I
     }
 
     public static String readPassword(LineReader passwordReader, String prompt) {
-        return passwordReader.readLine(prompt, (char) 0);
+        return passwordReader.readLine(prompt, '*');
```
Comment (jamesreprise) [inline common/Utils.java:76]: (char) 0 would make the prompt empty, making it unclear whether a password was being typed at all. Now, this asterisk is given in place. This does leak the length of the password being typed in exchange for some UX.

Comment (lolski) [reply common/Utils.java:76]: I agree that the UX would be better this way.

File: TypeDBConsole.java [hunk 2]
```diff
@@ -771,7 +772,7 @@ private static class CLIOptions implements Runnable {
         @CommandLine.Option(
                 names = {"--password"},
                 description = "Password",
-                prompt = "Enter password:",
+                prompt = "Password: ",
```
Comment (lolski) [inline TypeDBConsole.java:775]: What is the use of `prompt` if we're reading the password programmatically in the code with `Utils.readPassword(passwordReader, "Password: ");`?

Comment (jamesreprise) [reply TypeDBConsole.java:775]: This is specifically for the launching of Console from a shell and connecting to Cluster, not anything within console's REPL. The experience would look something like:
```
$ ./typedb console --cluster=localhost:1729 --username admin 
Password: ********
> transaction schema write [...]
```

**Review by lolski — APPROVED** [review]

File: TypeDBConsole.java (hunk 2 above)
Comment (lolski) [reply TypeDBConsole.java:775]: Oh right! Got it.

**Pull request merged**